cpp/src/arrow/compute/api_scalar.h

```diff
@@ -223,6 +223,18 @@ class ARROW_EXPORT SetLookupOptions : public FunctionOptions {
   bool skip_nulls;
 };
 
+/// Options for struct_field function
+class ARROW_EXPORT StructFieldOptions : public FunctionOptions {
+ public:
+  explicit StructFieldOptions(std::vector<int> indices);
```

Comment:
I wonder whether this should also accept a `FieldRef` or field resolution should be left to the caller.

Reply:
FieldRef is relative to a schema so we'd want/need a variant of this function that operates on a RecordBatch.

Reply:
But a plain string name could be useful for a StructArray?

Reply:
We'd still need a type to resolve it to an index, right? Unless you mean storing `std::vector<std::string>` or `std::string` directly? (Which might be reasonable.)

Reply:
Yes, storing it as strings on the options, I meant. Because when actually executing the kernel, the struct array itself can perfectly resolve the name I think?

Reply:
That's fair. Want to file a followup? I think we can support having a FieldRef internally, basically. (Though the interpretation will be a little different - it'll be relative to an array, not a schema.)

Reply:
Now that I think of it, it's probably better to resolve the field up front (using the schema) than pay the cost for every kernel invocation with the same schema.